src/adapter/src/coord/sequencer/inner/subscribe.rs

```diff
@@ -261,13 +261,8 @@ impl Coordinator {
                 return Err(AdapterError::AbsurdSubscribeBounds { as_of, up_to });
             }
         }
-        // Store read holds so that dependencies cannot be dropped during off-thread optimization.
```

Comment:
@jkosh44 We're changing how this works: now we only have one `ReadHolds` and we explicitly drop all of that below, instead of just popping off the last one. I'm curious to see how CI likes that and/or if this might come with other problems.

Reply:
I don't really understand the previous behavior to be honest. What were the other read holds for?

Reply:
I think it was just because Matt saw that we now had a Vec of ReadHolds, which Gábor only recently introduced, and then just went with it. AFAIK, there should be no other read holds and you can't mix SUBSCRIBE with other things in a transaction. @mjibson What do you think?

Reply:
Well, as long as CI is happy, I'll be happy.

Reply:
The changes in this file LGTM. The description above accurately describes what I did. The change here feels much better than depending on some element position in a Vec.

Reply:
thanks for taking the time!